### PR TITLE
add plan backend (model + api)

### PR DESCRIPTION
**Pull request by AlbertMartinezM**
closes issue #20 
Some required endpoints have not been created yet such as delete plan or ~~delete user~~ or get all plans from one user.

Endpoints created so far:

- /api/login 
-- POST : Creates user 
- /api/plan
-- POST: Creates new plan
- /api/plan/<string:id>
-- GET: returns id matching plan
-- PUT: updates plan. All fields have to be sent.
- /api/user:
-- GET: returns your user data
-- PUT: updates user
- /api/user/<string:uuid>:
-- GET: returns user that matches with uuid
-- PUT: update user. only username filed have to be provided

⚠️ config option JWT_COOKIE_CSRF_PROTECT is enabled by default (i can change it) and it requires to add X-CSRF-TOKEN header to every request with body

### CODE REVIEW

File: backend/exceptions/user_errors.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        try:
+            self.message = f'Could not create user obj using the following dict: {json.dumps(data)}'
+        except:
+            self.message = 'Could not create user obj using the provided dict'
+        super().__init__(self.message)
+
+
+class UserDBAddingError(Exception):
+    def __init__(self, user):
+        self.message = f'could not add the following user into the db:\n{json.dumps(user.json())}'
```
Comment (jorbush) [inline backend/exceptions/user_errors.py:23]: Capital letter is missing

File: backend/models/plan.py [hunk 2]
```diff
@@ -1,4 +1,118 @@
...
+        return {
+            'id': self.uid,
+            'name': self.name,
+            'description': self.description,
+            'date': self.date.strftime('%d/%m/%Y'),
+            'location': self.location,
+            #'admin': self.admin.json(),
+            'admin': self.admin,
+            'participants': self.participants
+            #'participants': [p.json() for p in self.participants]
```
Comment (jorbush) [inline backend/models/plan.py:30]: Are the comments relevants? If not, remove them pls

File: backend/models/plan.py [hunk 3]
```diff
@@ -1,4 +1,118 @@
...
+        p = db.collection(u'plans').document(id).get().to_dict()
+        if p is None:
+            raise PlanNotFoundError(id)
+        p['uid'] = id
+        try:
+            return Plan.from_dict(p)
+        except Exception as e:
+            raise PlanCreationError(p) from e
+
+    """
```
Comment (jorbush) [inline backend/models/plan.py:85]: Why is commented?

File: backend/models/plan.py (hunk 2 above)
Comment (AlbertMartinezM) [reply backend/models/plan.py:30]: idk which option would be better. will explain it in more detail at the weekly meeting

File: backend/models/plan.py (hunk 3 above)
Comment (AlbertMartinezM) [reply backend/models/plan.py:85]: currently working on get_user_plans(). Commented to clarify thay is not finished yet
add/remove participant will be finished in the appropiate PR

but yeah i can remove the comments and leave it as code

Comment (jorbush) [reply backend/models/plan.py:85]: it is not necessary

**Review by jorbush — APPROVED** [review]

**Pull request merged**